README.md

```diff
@@ -10,399 +10,451 @@ Welcome to A December of Algorithms. This is a small collection of algorithms to
+    - `Note #3:`Alternatively, you can look into saving your API key as an Environment Variable.
+    One final alternative is to simply store the API key in a text file and add that file to your `.gitignore` file. When you need to access it, open the file using your program and retrieve the API key.
+
+
+  - **Sample Output**
+    ```
+    Taken at 20:35 (IST), 04-12-2018
+    Current Temperature difference between Chennai,TN and London,GB is 19°C
+    ```
+  - **Resources:**
+    - [How to get your API key from OpenWeatherMap](https://openweathermap.org/appid)
+    - [How to retrieve weather data for a city](https://openweathermap.org/current#name)
 
 
-### **December 16 - Find Temperature Difference Between Two Cities**
-- **Problem**
-  - It’s getting harder to turn around in tech without bumping into some reference to APIs, or application programming interfaces. In the simplest terms, applications talk to each other via APIs.
-  - Usually, when we use an API to retrieve data (known as an API Call), we receive a JSON file containing data. To access this data, we will need to parse the JSON file. Depending on the language you are using, there are different ways to parse it. Just Google how to parse a JSON file in your language.
-  - OpenWeatherMap is a popular service that provides satellite images, Weather data, Historical data and Soil temperature and moisture.
-  - The problem is to find the current temperature difference between `London, GB` and `Chennai, Tamil Nadu, India` using the OpenWeatherMap API.
-  - `Note #1:` OpenWeatherMap APIs need a valid API key to allow responses, Packages won't work if you don't provide one. You can signup for a free API key on the OpenWeatherMap website.
-  - `Note #2:` API keys are private and **should not be made public**. When committing your code, please make sure you replace your original API key with a dummy key.
-  - `Note #2:`Alternatively, you can look into saving your API key as an Environment Variable.
-  One final alternative is to simply store the API key in a text file and add that file to your `.gitignore` file. When you need to access it, open the file using your program and retrieve the API key.
-
-
-- **Sample Output**
-  ```
-  Taken at 20:35 (IST), 04-12-2018
-  Current Temperature difference between Chennai,TN and London,GB is 19°C
-  ```
+### **December 17 - Magic Squares**
+  - **Problem**
+    - A magic square of order n is an arrangement of ``n^2`` numbers, containing distinct integers from ``1`` to ``n^2``, in a square, such that the ``n`` numbers in all rows, all columns, and both diagonals sum to the same constant.
+    - This constant is called the magic constant or magic sum, M, which is given by:
+        ```math
+        M = n(n^2+1)/2
+        ```
+    - Now the problem is to generate all the possible magic squares if the order of the square is given as input.
+
```

Comment:
`Hint:` It will be helpful if you breakdown the problem. First generate one magic square. Before generating the others, look at the other squares in the example carefully, you will notice something very useful.